src/plugins/ui_stateful/clipboard.ts

```diff
@@ -109,6 +109,40 @@ export class ClipboardPlugin extends UIPlugin {
           this.status = "invisible";
         }
         break;
+      case "REPLICATE_CELLS_ABOVE":
+        if (this.getters.getSelectedZones().length === 1) {
+          const zone = this.getters.getSelectedZone();
+          const singleRowSelected = zone.top !== zone.bottom;
+          const copyTarget = {
+            ...zone,
+            bottom: singleRowSelected ? zone.top : zone.top - 1,
+            top: singleRowSelected ? zone.top : zone.top - 1,
+          };
+          const state = this.getClipboardState([copyTarget], "COPY");
+          state.paste([zone], {
```

Comment:
spawning a side clipboard state out of the blue feels weird

Reply:
@rrahir Earlier, I reused `ClipboardCellState` but then @hokolomopo pointed out that I should not be changing it's state otherwise it would mess things up in case of figure. I too, realised the same and hence I used the state as discussed [here](https://github.com/odoo/o-spreadsheet/pull/2818#discussion_r1318175545).